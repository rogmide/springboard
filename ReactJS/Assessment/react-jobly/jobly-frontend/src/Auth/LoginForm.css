.LoginForm {
  display: flex;
  flex-direction: column;
  width: 300px;
  margin-top: 10%;
  border: 1px solid #343a40;
  border-radius: 10px;
  padding: 20px;
}
