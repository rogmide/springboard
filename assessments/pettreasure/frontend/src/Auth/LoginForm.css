.LoginForm {
  display: flex;
  flex-direction: column;
  width: 300px;
  margin-top: 10%;
  border: 0.5px solid #acb5bf;
  border-radius: 10px;
  padding: 20px;
}
