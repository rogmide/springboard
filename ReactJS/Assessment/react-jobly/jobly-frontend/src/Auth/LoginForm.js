import React, { useState } from "react";
import "./LoginForm.css";

const LoginForm = ({ login }) => {
  const INITIAL_STATE = {
    username: "",
    password: "",
  };
  const [formData, setFormData] = useState(INITIAL_STATE);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((data) => ({
      ...data,
      [name]: value,
    }));
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    login({ ...formData });
    setFormData(INITIAL_STATE);
  };

  return (
    <>
      <div className="container LoginForm">
        <h1>Login</h1>
        <form onSubmit={handleSubmit}>
          <div className="form-group">
            <label htmlFor="username">Username: </label>
            <input
              className="form-control"
              required
              id="username"
              type="text"
              name="username"
              placeholder="Username"
              value={formData.username}
              onChange={handleChange}
            ></input>
          </div>
          <label htmlFor="password">Password: </label>
          <input
            className="form-control"
            required
            id="password"
            type="password"
            name="password"
            placeholder="Password"
            value={formData.password}
            onChange={handleChange}
          ></input>
          <br></br>
          <button className="btn btn-secondary">Submit</button>
        </form>
      </div>
    </>
  );
};

export default LoginForm;
