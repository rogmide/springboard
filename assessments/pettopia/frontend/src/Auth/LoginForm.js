import React, { useState } from "react";
import "./LoginForm.css";
import { Navigate } from "react-router-dom";

const LoginForm = ({ login }) => {
  const INITIAL_STATE = {
    username: "",
    password: "",
  };
  const [formData, setFormData] = useState(INITIAL_STATE);
  const [error, setError] = useState("");

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((data) => ({
      ...data,
      [name]: value,
    }));
  };

  async function handleSubmit(e) {
    e.preventDefault();
    const resp = await login({ ...formData });
    setError(resp);
  }

  return (
    <>
      <div className="container LoginForm">
        <h1>Login</h1>
        {error === true ? (
          <Navigate exact="true" to="/" />
        ) : (
          <h5 style={{ color: "red", fontSize: "18px", textAlign: "center" }}>
            {error}
          </h5>
        )}
        <form onSubmit={handleSubmit}>
          <div className="form-group">
            <label htmlFor="username">Username: </label>
            <input
              className="form-control"
              required
              id="username"
              type="text"
              name="username"
              placeholder="Username"
              value={formData.username}
              onChange={handleChange}
            ></input>
          </div>
          <label htmlFor="password">Password: </label>
          <input
            className="form-control"
            required
            id="password"
            type="password"
            name="password"
            placeholder="Password"
            value={formData.password}
            onChange={handleChange}
          ></input>
          <br></br>
          <button className="btn btn-secondary float-right">Submit</button>
        </form>
      </div>
    </>
  );
};

export default LoginForm;
