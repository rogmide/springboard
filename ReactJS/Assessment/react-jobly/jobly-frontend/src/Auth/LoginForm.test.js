import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import LoginForm from "./LoginForm";

// SNAPSHOT TEST
test("should match snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <LoginForm />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
