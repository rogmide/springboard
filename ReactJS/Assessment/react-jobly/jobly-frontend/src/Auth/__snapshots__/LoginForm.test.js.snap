// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="container LoginForm"
  >
    <h1>
      Login
    </h1>
    <h5
      style="color: red; font-size: 18px; text-align: center;"
    />
    <form>
      <div
        class="form-group"
      >
        <label
          for="username"
        >
          Username: 
        </label>
        <input
          class="form-control"
          id="username"
          name="username"
          placeholder="Username"
          required=""
          type="text"
          value=""
        />
      </div>
      <label
        for="password"
      >
        Password: 
      </label>
      <input
        class="form-control"
        id="password"
        name="password"
        placeholder="Password"
        required=""
        type="password"
        value=""
      />
      <br />
      <button
        class="btn btn-secondary float-right"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
